controllers/provider-credential-controller.go

```diff
@@ -86,7 +96,7 @@ func (r *ProviderCredentialSecretReconciler) Reconcile(ctx context.Context, req
 	log.V(1).Info("NEW Provider hash: " + hex.EncodeToString(currentHash))
 
 	// If no hash is found, store the currentHash (this is for NEW or MIGRATED Provider Secrets)
-	if originalHash == nil {
+	if a == nil {
```

Comment:
Maybe you want to keep the test on originalHash check, just in case the decode failed, you would write a new version or maybe better if the decode fails, we should stop and throw an error for the user to fix it?

Reply:
we should throw error in the logs, and also will check the originalHash